rpc/contract.go

```diff
@@ -39,6 +39,13 @@ func (h *Handler) StorageAt(address, key felt.Felt, id BlockID) (*felt.Felt, *js
 	}
 	defer h.callAndLogErr(stateCloser, "Error closing state reader in getStorageAt")
 
+	// This checks if the contract exists because if a key doesn't exist in contract storage,
+	// the returned value is always zero and error is nil.
+	_, err := stateReader.ContractNonce(&address)
+	if err != nil {
```

Comment:
We already discussed it in slack, but I think it's worth to discuss it here again:
do you think that we should handle non ErrKeyNotFound errors differently? i.e.:
```
_, err := stateReader.ContractNonce(&address)
if err != nil {
   if errors.Is(err, db.ErrKeyNotFound) {
       return nil, ErrContractNotFound
   } else {
       s.log.Errorw("Failed to get contract nonce", "err", err)
       return nil, ErrInternal
   }
}
```
?

Reply:
Yeah this is better, makes it more robust.